cheroot/server.py

```diff
@@ -1204,6 +1254,190 @@ def send_headers(self):
         self.conn.wfile.write(EMPTY.join(buf))
 
 
+class HyperHTTPRequest(HTTPRequest):
+    """An HTTP Request (and response).
+
+    A single HTTP connection may consist of multiple request/response pairs.
+    """
+
+    server = None
+    """The HTTPServer object which is receiving this request."""
+
+    conn = None
+    """The HTTPConnection object on which this request connected."""
+
+    inheaders = {}
+    """A dict of request headers."""
+
+    outheaders = []
+    """A list of header tuples to write in the response."""
+
+    ready = False
+    """When True, the request has been parsed and is ready to begin generating
+    the response. When False, signals the calling Connection that the response
+    should not be generated and the connection should close."""
+
+    close_connection = False
+    """Signals the calling Connection that the request should close. This does
+    not imply an error! The client and/or server may each request that the
+    connection be closed."""
+
+    chunked_write = False
+    """If True, output will be encoded with the "chunked" transfer-coding.
+
+    This value is set automatically inside send_headers."""
+
+    header_reader = HeaderReader()
+    """
+    A HeaderReader instance or compatible reader.
+    """
+
+    def __init__(self, server, conn, proxy_mode=False, strict_mode=True):
+        """Initialize HTTP request container instance.
+
+        Args:
+            server (HTTPServer): web server object receiving this request
+            conn (HTTPConnection): HTTP connection object for this request
+            proxy_mode (bool): whether this HTTPServer should behave as a PROXY
+            server for certain requests
+            strict_mode (bool): whether we should return a 400 Bad Request when
+            we encounter a request that a HTTP compliant client should not be
+            making
+        """
+        self.server = server
+        self.conn = conn
+        self.h_conn = h11.Connection(h11.SERVER)
+        self.ready = False
+        self.started_request = False
+        self.scheme = b'http'
+        if self.server.ssl_adapter is not None:
+            self.scheme = b'https'
+        # Use the lowest-common protocol in case read_request_line errors.
+        self.response_protocol = 'HTTP/1.0'
+        self.inheaders = {}
+
+        self.status = ''
+        self.outheaders = []
+        self.sent_headers = False
+        self.close_connection = self.__class__.close_connection
+        self.chunked_read = False
+        self.chunked_write = self.__class__.chunked_write
+        self.proxy_mode = proxy_mode
+        self.strict_mode = strict_mode
+
+    def _next_event(self):
```

Comment:
This name doesn't add value to the semantics of its purpose. Method names should start with verbs because methods/functions ultimately represent actions.
Also, it seems to be called once, "next" in the name kinda suggests that it's supposed to be called multiple times which is misleading.

Let's think of something better here. What does it do? To me, it looks like it `await`s for some event to happen and also communicates some data/state to the other end.

So maybe something like `_communicate_100`? But it's probably too much implementation details also it's not the primary purpose of this method, is it?

How about
```suggestion
    def _await_request_line_bytes(self):
```
?

Reply:
I have renamed it to `_process_next_h11_event`. This seems to be the most accurate; depending on the execution, it may read data into h11's internal buffer, then attempt to get an event from h11. 

There is another function that calls this method that does not read and simply asks h11 if any new events have been queued since the last time an event was produced.